Scala/Application.scala

```diff
@@ -0,0 +1,169 @@
+
+import play.modules.reactivemongo._
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api._
+import play.api.mvc._
+import models.movies
+import reactivemongo.play.json.collection.JSONCollection
+import reactivemongo.play.json._
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import reactivemongo.api._
+import scala.concurrent.Future
+import play.api.libs.json._
+
+
+class Application @Inject()(val reactiveMongoApi: ReactiveMongoApi, val messagesApi: MessagesApi) extends Controller with I18nSupport with MongoController with ReactiveMongoComponents{
+
+// Gets a collection from a Mongo DB ('movies')
+  def collection: Future[JSONCollection] = database.map(_.collection[JSONCollection]("movies"))
+
+// Goes to the Home Page
+  def index = Action {
+    Ok(views.html.index("Test Output", "Welcome to the home page!"))
+  }
+
+// Adds a new movie to the database, then goes to the View Page to display it.
+  def toView = Action { implicit request: Request[AnyContent] =>
+    val formValidationResult = movies.createMovieForm.bindFromRequest
+    formValidationResult.fold({
+      formWithErrors =>
+        BadRequest(views.html.theView(movies.theMovies, formWithErrors))
+    }, { widget =>
+      widget.mID = movies.theID
+      collection.flatMap(_.insert(widget))
+      Redirect(routes.Application.listMovies)
+    })
+  }
+
+  // Goes to the View Page (view list of movies)
+  def listMovies = Action.async {
+
+    val cursor: Future[Cursor[movies]] = collection.map {
+      _.find(Json.obj()).cursor[movies]
+    }
+
+    val futureMoviesList: Future[List[movies]] = cursor.flatMap(_.collect[List]())
+    futureMoviesList.map { loadMovies =>
+      movies.theMovies.clear()
+      for(x <- loadMovies)
+        {
+          movies.theMovies += x
+        }
+
+      movies.theID
+      Ok(views.html.theView(movies.theMovies, movies.createMovieForm))
+    }
+
+  }
+
+
+// Goes to the Edit Page, sending the ID of the movie the user wants to edit.
+  def toEdit(theID:Int) = Action {
+    val oneMovie = movies.theMovies.zipWithIndex.filter(_._1.mID == theID).head._2
+    Ok(views.html.theEdit(movies.theMovies, movies.createMovieForm.fill(movies.theMovies(oneMovie)), oneMovie))
+  }
+
+ // Edits an existing movie in the Mongo DB, then goes to the View Page to display the edit.
+  def doEdit = Action { implicit request: Request[AnyContent] =>
+    val formValidationResult = movies.createMovieForm.bindFromRequest
+    formValidationResult.fold({
+      formWithErrors =>
+        BadRequest(views.html.theView(movies.theMovies, formWithErrors))
+    }, { widget =>
+
+      val modifier = Json.obj("$set" -> Json.obj("title"->widget.title,
+        "description"->widget.description,
+        "price"->widget.price,
+        "director" -> widget.director,
+        "starring" -> widget.starring,
+        "rating" -> widget.rating
+      ))
+
+      collection.map{
+        _.findAndUpdate(Json.obj("mID"->widget.mID),modifier)
```

Comment:
Can't you just directly pass in widget as a whole instead of modifier and manually doing the JSON mapping? I could swear you should be able to do it